### PR TITLE
[CELEBORN-1111][DOC][FOLLOWUP] Add HDFS Kerberos usage doc

### PR DESCRIPTION
**Pull request by liujiayi771**
### What changes were proposed in this pull request?
Add HDFS Kerberos usage doc.

### CODE REVIEW

File: docs/deploy.md [hunk 1]
```diff
@@ -59,6 +59,15 @@ celeborn.worker.flusher.hdfs.buffer.size 4m
...
 celeborn.worker.replicate.fastFail.duration 240s
 
+# If your HDFS is enabled with Kerberos authentication. You also need to set the following
+# parameters for Celeborn master and worker. You need to deploy the keytab file to the
+# corresponding path on the master and worker.
+celeborn.storage.hdfs.kerberos.enabled true
+celeborn.storage.hdfs.kerberos.principal user@REALM
+celeborn.storage.hdfs.kerberos.keytab /path/to/user.keytab
+celeborn.hadoop.hadoop.security.authorization kerberos
+celeborn.hadoop.dfs.namenode.kerberos.principal hdfs/_HOST@REALM
```
Comment (pan3793) [inline docs/deploy.md:69]: in practice, these hadoop conf are usually set in `core-site.xml` or `hdfs-site.xml`, does export `HADOOP_CONF_DIR` works?

Comment (liujiayi771) [reply docs/deploy.md:69]: No. It seems that Celeborn create a new hadoop configuration, and not load configuration files in `HADOOP_CONF_DIR`.

Comment (pan3793) [reply docs/deploy.md:69]: this is an unintuitive behavior

Comment (waitinfuture) [reply docs/deploy.md:69]: cc @FMX please verify this

**Comment by FMX** [comment]
As I have changed the way Celeborn supports Kerberos. I think this PR should be closed.

**Comment by liujiayi771** [comment]
> As I have changed the way Celeborn supports Kerberos. I think this PR should be closed.

Not all people have Celeborn and HDFS in the same cluster, we should keep the configs, and also support use cluster `core-site.xml `and `hdfs-site.xml`

**Pull request closed without merging**